tests/change_detector/change_detector_test.go

```diff
@@ -0,0 +1,132 @@
+// Copyright 2023 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+//go:build change_detector
+
+package change_detector
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestChanges(t *testing.T) {
+	sourceRepoDir := t.TempDir()
+	execClone(t, sourceRepoDir, repository, sourceBranch)
+
+	if checkIfDatabaseFormatChangesAreDocumented(t, sourceRepoDir) {
+		t.Skip("skipping test with documented database format changes")
+	}
+
+	var targetRepoDir string
+	if targetBranch == "" {
+		// default to the local branch
```

Comment:
question: This seems odd and a bit dangerous no?  If I understand correctly this will now default to testing against itself, which is essentially the same as running `go test ./...`?

This does seem like an easy way to accidentally not run the change detector at all.